src/libbluechi/socket.c

```diff
@@ -39,7 +38,7 @@ int create_tcp_socket(uint16_t port) {
                 return -errsv;
         }
 
-        return steal_fd(&fd);
+        return fd;
```

Comment:
Please revert this change, having `steal_fd` here is intentional: https://github.com/containers/bluechi/pull/602